### PR TITLE
✨ Bump golangci lint to v1.49.0

### PR DESCRIPTION
**Pull request by sbueringer**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -132,6 +132,9 @@ issues:
   - linters:
     - gosec
     text: "G304: Potential file inclusion via variable"
+  - linters:
+    - revive
+    text: "package-comments: should have a package comment"
```
Comment (sbueringer) [inline .golangci.yml:137]: If we want to add package comments everywhere I would suggest to do that in follow-up PRs (I can open an issue for it).

There are a *lot* of packages with missing package comments

**Comment by sbueringer** [comment]
There are 4 open godot findings. In my opinion all false positives.

I see the following options:
* drop godot
* add a dot to the last elements of the lists

**Comment by alvaroaleman** [comment]
> drop godot

I'd be in favor of that

**Comment by sbueringer** [comment]
> I'd be in favor of that

Done. Not sure what happened. Tried to do a commit -amend with GitHub codespaces and that somehow closed the PR :/

New PR here: https://github.com/kubernetes-sigs/controller-runtime/pull/1988/files

**Pull request closed without merging**